Mathlib/RingTheory/LocalProperties/Exactness.lean

```diff
@@ -0,0 +1,214 @@
+  [AddCommMonoid M] [Module R M] [Module A M] [AddCommMonoid N] [Module R N] [Module A N]
+
+variable
+  (Mₚ : ∀ (P : Ideal R) [P.IsMaximal], Type*)
+  [∀ (P : Ideal R) [P.IsMaximal], AddCommMonoid (Mₚ P)]
+  [∀ (P : Ideal R) [P.IsMaximal], Module R (Mₚ P)]
+  (f : ∀ (P : Ideal R) [P.IsMaximal], M →ₗ[R] Mₚ P)
+  [∀ (P : Ideal R) [P.IsMaximal], IsLocalizedModule P.primeCompl (f P)]
+  (Nₚ : ∀ (P : Ideal R) [P.IsMaximal], Type*)
+  [∀ (P : Ideal R) [P.IsMaximal], AddCommMonoid (Nₚ P)]
+  [∀ (P : Ideal R) [P.IsMaximal], Module R (Nₚ P)]
+  (g : ∀ (P : Ideal R) [P.IsMaximal], N →ₗ[R] Nₚ P)
+  [∀ (P : Ideal R) [P.IsMaximal], IsLocalizedModule P.primeCompl (g P)]
+  (k : M →ₗ[R] N)
+
+theorem injective_of_isLocalized_maximal
+    (H : ∀ (P : Ideal R) [P.IsMaximal], Function.Injective (map P.primeCompl (f P) (g P) k)) :
+    Function.Injective k :=
+  fun x y eq ↦ Module.eq_of_localization_maximal _ f _ _ fun P _ ↦ H P <| by simp [eq]
+
+theorem surjective_of_isLocalized_maximal
+    (H : ∀ (P : Ideal R) [P.IsMaximal], Function.Surjective (map P.primeCompl (f P) (g P) k)) :
+    Function.Surjective k :=
+  range_eq_top.mp <| eq_top_of_localization₀_maximal Nₚ g _ <|
+    fun P _ ↦ (range_localizedMap_eq_localized₀_range _ (f P) (g P) k).symm.trans <|
+    range_eq_top.mpr <| H P
+
+theorem bijective_of_isLocalized_maximal
+    (H : ∀ (P : Ideal R) [P.IsMaximal], Function.Bijective (map P.primeCompl (f P) (g P) k)) :
+    Function.Bijective k :=
+  ⟨injective_of_isLocalized_maximal Mₚ f Nₚ g k fun J _ ↦ (H J).1,
+  surjective_of_isLocalized_maximal Mₚ f Nₚ g k fun J _ ↦ (H J).2⟩
+
+variable {L : Type*} [AddCommMonoid L] [Module R L] [Module A L]
+variable (Lₚ : ∀ (P : Ideal R) [P.IsMaximal], Type*)
+  [∀ (P : Ideal R) [P.IsMaximal], AddCommMonoid (Lₚ P)]
+  [∀ (P : Ideal R) [P.IsMaximal], Module R (Lₚ P)]
+  (h : ∀ (P : Ideal R) [P.IsMaximal], L →ₗ[R] Lₚ P)
+  [∀ (P : Ideal R) [P.IsMaximal], IsLocalizedModule P.primeCompl (h P)]
+
+theorem exact_of_isLocalized_maximal (F : M →ₗ[R] N) (G : N →ₗ[R] L)
+    (H : ∀ (J : Ideal R) [J.IsMaximal],
+      Function.Exact (map J.primeCompl (f J) (g J) F) (map J.primeCompl (g J) (h J) G)) :
+    Function.Exact F G := by
+  simp only [LinearMap.exact_iff] at H ⊢
+  apply eq_of_localization₀_maximal Nₚ g
+  intro J hJ
+  rw [← LinearMap.range_localizedMap_eq_localized₀_range _ (f J) (g J) F,
+    ← LinearMap.ker_localizedMap_eq_localized₀_ker J.primeCompl (g J) (h J) G]
+  have := SetLike.ext_iff.mp <| H J
+  ext x
+  simp only [mem_range, mem_ker] at this ⊢
+  exact this x
+
+end isLocalized_maximal
+
+section localized_maximal
+
+variable {R M N : Type*} [CommSemiring R] [AddCommMonoid M] [Module R M]
+  [AddCommMonoid N] [Module R N] (f : M →ₗ[R] N)
+
+theorem injective_of_localized_maximal
+    (h : ∀ (J : Ideal R) [J.IsMaximal], Function.Injective (map J.primeCompl f)) :
+    Function.Injective f :=
+  injective_of_isLocalized_maximal
+    _ (fun P _ ↦ mkLinearMap P.primeCompl M)
```

Comment:
Looks like this works everywhere `mkLinearMap` is called:
```suggestion
    _ (fun _ _ ↦ mkLinearMap _ _)
```
but it's up to you whether to replace them.